python/tank/bootstrap/configuration.py

```diff
@@ -111,14 +114,18 @@ def get_tk_instance(self, sg_user):
 
         return tk
 
-    def _set_authenticated_user(self, user):
+    def _set_authenticated_user(self, user, serialized_user):
```

Comment:
i think we should just pass the serialized user here.

Reply:
We need to pass both. If unserialization fails, we need to set back the original user just so we can keep going. I've tried rewriting this without having to pass the original user in and have the method return a boolean indicating if the user could be set, but the code was becoming a mess and it made it harder to test so I've prefered to keep it that way.